commands/inspect.go

```diff
@@ -0,0 +1,240 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package commands
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/openpubkey/openpubkey/pktoken"
+	"golang.org/x/crypto/ssh"
+)
+
+type InspectCmd struct {
+	// KeyOrCert is the SSH key or certificate to be inspected.
+	KeyOrCert string
+	// Output is where output should be written to.
+	Output io.Writer
+}
+
+// NewInspectCmd creates a new InspectCmd instance with the provided arguments.
+func NewInspectCmd(keyOrCert string, output io.Writer) *InspectCmd {
+	return &InspectCmd{
+		KeyOrCert: keyOrCert,
+		Output:    output,
+	}
+}
+
+// printf formats a string to the configured output.
+func (i *InspectCmd) printf(format string, a ...any) {
+	if _, err := fmt.Fprintf(i.Output, format, a...); err != nil {
+		// Fall back to stdout
+		i.printf(format, a...)
+	}
+}
+
+func (i *InspectCmd) Run(ctx context.Context) error {
+	// Check if the input is a file path
+	if _, err := os.Stat(i.KeyOrCert); err == nil {
+		// It's a file, read its contents
+		data, err := os.ReadFile(i.KeyOrCert)
+		if err != nil {
+			return fmt.Errorf("error reading input file: %v", err)
+		}
+		i.KeyOrCert = string(data)
+	}
+
+	// Trim whitespace and newlines
+	i.KeyOrCert = strings.TrimSpace(i.KeyOrCert)
+
+	// Parse the SSH key or certificate
+	pubKey, _, _, _, err := ssh.ParseAuthorizedKey([]byte(i.KeyOrCert))
+	if err != nil {
+		return fmt.Errorf("failed to parse SSH key: %v", err)
+	}
+
+	// Check if it's a certificate
+	if cert, ok := pubKey.(*ssh.Certificate); ok {
+		i.inspectCertificate(cert)
+	} else {
+		// It's a regular public key
+		i.inspectPublicKey(pubKey)
+	}
+
+	return nil
+}
+
+func (i *InspectCmd) inspectCertificate(cert *ssh.Certificate) {
+	i.printf("--- SSH Certificate Information ---\n")
+	i.printf("%-18s %d\n", "Serial:", cert.Serial)
+	i.printf("%-18s %s\n", "Type:", certificateType(cert.CertType))
+	i.printf("%-18s %s\n", "Key ID:", cert.KeyId)
+	i.printf("%-18s %v\n", "Principals:", cert.ValidPrincipals)
+	i.printf("%-18s %s\n", "Valid After:", formatTime(cert.ValidAfter))
+	i.printf("%-18s %s\n", "Valid Before:", formatTime(cert.ValidBefore))
+	i.printf("%-18s %v\n", "Critical Options:", cert.CriticalOptions)
+
+	// Format extensions nicely
+	i.printf("Extensions:\n")
+	for key, value := range cert.Extensions {
+		if key == "openpubkey-pkt" {
+			i.printf("  %s: [PKToken data]\n", key)
```

Comment:
Add the number of bytes in PK Token data.

```suggestion
			i.printf("  %s: [PKToken data] %d bytes \n", key, len(value))
```

This might save someone an afternoon of debugging where they think openpubkey-pkt contains the PK Token but it is actually empty.

Reply:
Good call!